ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -23,3 +26,92 @@ def __init__(self, resource_name="", *args, **kwargs):
         super(StorageCluster, self).__init__(
             resource_name=resource_name, kind='StorageCluster', *args, **kwargs
         )
+
+
+def count_cluster_osd():
+    """
+    The function returns the number of cluster OSDs
+
+    Returns:
+         osd_count (int): number of OSD pods in current cluster
+
+    """
+    storage_cluster_obj = StorageCluster(
+        resource_name=config.ENV_DATA['storage_cluster_name'],
+        namespace=config.ENV_DATA['cluster_namespace'],
+    )
+    storage_cluster_obj.reload_data()
+    osd_count = (
+        int(storage_cluster_obj.data['spec']['storageDeviceSets'][0]['count'])
+        * int(storage_cluster_obj.data['spec']['storageDeviceSets'][0]['replica'])
+    )
+    return osd_count
+
+
+def add_capacity(capacity_string):
+    """
+   Add storage capacity to the cluster
+
+   Args:
+       capacity_string(int): Size of the storage to add
+   Returns:
+       True if capacity was added, False if not
+
+   """
+    ocp = OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE, kind=constants.STORAGECLUSTER)
+    old_osd_count = count_cluster_osd()
+    osd_size = parse_size_to_int(get_storage_cluster()['items'][0]['spec']['storageDeviceSets'][0]['dataPVCTemplate']
+                                                      ['spec']['resources']['requests']['storage'])
+    capacity_to_add = parse_size_to_int(capacity_string/osd_size + old_osd_count)
```

Comment:
do not use old_osd_count, instead you should use 'existing count of storageDeviceSets'. get this count before running 'patch' command.

Correct formula = (capacity requested / osd capacity ) + existing count storageDeviceSets

Reply:
The capacity adding Im doing by changing the device set count, In the validation Im checking that  a correct amount of OSDs was added according to the new device set count

Reply:
for now this verification was removed. checking only cluster health